### PR TITLE
Added support for _elements search parameter

### PR DESCRIPTION
**Pull request by anyname2**
## Description
Added support for `_elements` search parameter: https://www.hl7.org/fhir/search.html#elements

## Testing
Added unit tests and E2E test. Also verified manually with SqlServer.

### CODE REVIEW

File: test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/BasicSearchTests.cs [hunk 1]
```diff
@@ -257,6 +257,32 @@ public async Task GivenResources_WhenSearchedWithIncorrectSummaryParams_ThenExce
             Assert.Equal(HttpStatusCode.BadRequest, ex.StatusCode);
         }
 
+        [Fact]
+        [Trait(Traits.Priority, Priority.One)]
+        public async Task GivenListOfResources_WhenSearchedWithElements_ThenOnlySpecifiedPropertiesShouldBeReturned()
```
Comment (brendankowitz) [inline test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/BasicSearchTests.cs:262]: Would be good to have some negative/empty/invalid cases too

Comment (anyname2) [reply test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/BasicSearchTests.cs:262]: I'm a bit preoccupied with another project. Can add some more tests towards the end of the week

Comment (anyname2) [reply test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/BasicSearchTests.cs:262]: I've added negative tests

**Pull request merged**